From 8481a5a0e006e4a2604451221cce2fa661f1e053 Mon Sep 17 00:00:00 2001
From: Ed Lee <edilee@mozilla.com>
Date: Wed, 2 Oct 2019 17:54:11 +0000
Subject: [PATCH 1/2] Bug 1585672 - [mach] Skip modules whose __file__
 attribute is None in the global ImportHook r=ahal

This also gets around a flake8 lint exception.

Differential Revision: https://phabricator.services.mozilla.com/D47946

--HG--
extra : moz-landing-system : lando

(cherry picked from commit b34af2b5813bd6500fafaa57d0361fc3a23dc7ad)
---
 build/mach_bootstrap.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/mach_bootstrap.py b/build/mach_bootstrap.py
index 4fb5981d0a4d..e341d1778180 100644
--- a/build/mach_bootstrap.py
+++ b/build/mach_bootstrap.py
@@ -480,7 +480,7 @@ class ImportHook(object):
         self._modules.add(resolved_name)
 
         # Builtin modules don't have a __file__ attribute.
-        if not hasattr(module, '__file__'):
+        if not getattr(module, '__file__', None):
             return module
 
         # Note: module.__file__ is not always absolute.
-- 
GitLab


From ad9d0b17d074fdd0dd8f7995ead4e27e66f07eb3 Mon Sep 17 00:00:00 2001
From: Marco Castelluccio <mcastelluccio@mozilla.com>
Date: Thu, 14 May 2020 18:28:55 +0000
Subject: [PATCH 2/2] Bug 1637911 - Skip modules with a None __file__
 attribute. r=rstewart

Differential Revision: https://phabricator.services.mozilla.com/D75267

(cherry picked from commit d845cf5481629e98c42027a51b065395b243033c)
---
 python/mozbuild/mozbuild/pythonutil.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/python/mozbuild/mozbuild/pythonutil.py b/python/mozbuild/mozbuild/pythonutil.py
index 918eb866d179..fc781a5aa520 100644
--- a/python/mozbuild/mozbuild/pythonutil.py
+++ b/python/mozbuild/mozbuild/pythonutil.py
@@ -11,8 +11,8 @@ import sys
 def iter_modules_in_path(*paths):
     paths = [os.path.abspath(os.path.normcase(p)) + os.sep
              for p in paths]
-    for name, module in list(sys.modules.items()):
-        if not hasattr(module, '__file__'):
+    for name, module in sys.modules.items():
+        if getattr(module, '__file__', None) is None:
             continue
 
         path = module.__file__
-- 
GitLab

